core/src/main/java/bio/terra/pearl/core/service/workflow/RegistrationService.java

```diff
@@ -125,6 +131,32 @@ private RegistrationResult register(String portalShortcode, EnvironmentName envi
         return new RegistrationResult(user, ppUser);
     }
 
+    public RegistrationResult registerGovernedUser(String portalShortcode, ParticipantUser proxy) {
+        PortalEnvironment portalEnv = portalEnvService.findOne(portalShortcode, proxy.getEnvironmentName()).get();
+        ParticipantUser mainUser = new ParticipantUser();
+        mainUser.setEnvironmentName(proxy.getEnvironmentName());
+        String mainUserName = proxy.getUsername() + "-proxied-";
+        String guid = generateGUID(mainUserName, proxy.getEnvironmentName());
+        mainUser.setUsername(mainUserName + guid);
+        mainUser = participantUserService.create(mainUser);
+
+        PortalParticipantUser ppUser = new PortalParticipantUser();
+        ppUser.setPortalEnvironmentId(portalEnv.getId());
+        ppUser.setParticipantUserId(mainUser.getId());
+
+        Profile profile = new Profile();
```

Comment:
for this and the other constructions, use the Builders
```suggestion
        Profile profile =  Profile.builder() 
           .contactEmail(...)
           ...
           .build()
```

Reply:
Don't forget this comment!